### DISCUSSION TITLE
Enable only Copilot with Internet access in the code

### DISCUSSION THREAD

**Discussion by BokDan93** [xtekky/gpt4free#1733] [Q&A]
Friends, tell me how to use the gpt4 or gpt4turbo model in the code via Copilot with Internet access enabled, so that the response receives up-to-date information from the Internet?

from g4f.client import Client

client = Client()
response = client.chat.completions.create(
    model="gpt-3.5-turbo",
    messages=[{"role": "user", "content": "Hello"}],
    ...
)
print(response.choices[0].message.content)

**Comment by hlohaus** [comment]
You can add , web_search=True, when you're using gpt-4 with the Bing provider.

**Comment by BokDan93** [comment]
Thanks for the answer, but I am a little at a loss in setting this parameter. could you please point or write some of the working code? Or leave a link to documentation on this issue.

Here is my code, I set the parameter, but when requesting I still can’t use the Internet

clientFree = Client()
                    loop = asyncio.get_event_loop()
                    response = await loop.run_in_executor(None, lambda: clientFree.chat.completions.create(
                            model="gpt-4-turbo",
                            messages=dialogue_history,
                            web_search=True
                    ))

**Reply by hlohaus** [reply]
How do you want use the internet?

Bing have 7 modes:

tone="Balance,Persise,Creative or Copilot",
web_search=True or False,

The Copilot Mode supports only web_search enabled.

Bing use only the web_search, when it thinks it is useful.

**Comment by BokDan93** [comment]
Thanks for the help! It was possible to set the desired setting
